use static page for broker liveness probe

pgier opened the pull request:
This brings the broker liveness probe in sync with the community Helm chart, and should use less resources than the health_check script.

pgier commented:
@cdbartholomew PTAL.  This is how the Apache community broker is configured and we're doing the same in Astra Streaming.

nicoloboschi reviewed (approved):
LGTM

michaeljmarshall reviewed (commented):
The tradeoff is that when the broker is deadlocked, k8s will no longer restart the pod. The current health check fails when there is deadlock.

pgier commented:
In Astra we were previously seeing brokers regularly restarting because the health check was failing, possibly incorrectly (maybe @zzzming has more info) when using the health check.  So we switched to use the metrics endpoint, but then we were seeing brokers stuck in a running but not ready state.  That issue seems much better in the current 2.10 versions that we're running.  We switched to using the static page instead of the metrics endpoint in Astra, and it seems to be fine for the past couple weeks.

michaeljmarshall commented:
Do we have an issue opened in the upstream project? That sounds like a bug.

pgier commented:
@michaeljmarshall I think the issue was fixed in 2.10.  At least we haven't seen it in the last couple months.  Maybe we need a new endpoint specific to the liveness check?

michaeljmarshall commented:
A dedicated liveness check could make sense. We'd just need to find the right things to check. I thought about this a few months ago, but I didn't come up with a good solution. Maybe it is worth a discussion on the dev list to ask "when is a broker alive and when is it ready?"
